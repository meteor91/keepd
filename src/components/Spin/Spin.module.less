.spinNested {
  position: relative;
  min-height: 50px;
  .spin {
    position: absolute;
    top: 0;
    left: 0;
    display: flex;
    height: 100%;
    width: 100%;
    justify-content: center;
    align-items: center;
    text-align: center;

    .loader {
      width: 25px;
      height: 25px;
      border: 5px solid #021f2638;
      border-top-color: #021f26;
      border-radius: 50%;
      animation: spinning 1s linear infinite;
    }

    @keyframes spinning {
      to {
        transform: rotate(360deg);
      }
    }
  }
  .contentBlur {
    opacity: 0.5;
    user-select: none;
    pointer-events: none;
  }
}



//.spin {
//  display: flex;
//  height: 100%;
//  width: 100%;
//  justify-content: center;
//  align-items: center;
//  text-align: center;
//
//  .loader {
//    width: 100px;
//    height: 100px;
//    border: 5px solid #021f2638;
//    border-top-color: #021f26;
//    border-radius: 50%;
//    animation: spinning 1s linear infinite;
//  }
//
//  @keyframes spinning {
//    to {
//      transform: rotate(360deg);
//    }
//  }
//}
