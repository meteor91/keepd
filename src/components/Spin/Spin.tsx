import React from 'react';
import clsx from 'clsx';

import style from './Spin.module.less';

export interface SpinProps {
    spinning?: boolean;
    children?: React.ReactNode;
}

export const Spin: React.FC<SpinProps> = (props) => {
    const { spinning } = props;
    return (
        <div className={style.spinNested}>
            {spinning && (
                <div className={clsx({ [style.spin]: spinning })} data-testid="spinner">
                    <div className={style.loader}/>
                </div>
            )}
            <div className={clsx({ [style.contentBlur]: spinning })} data-testid="spin-content">
                {props.children}
            </div>
        </div>
    );
};
