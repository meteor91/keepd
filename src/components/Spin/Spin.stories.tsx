import React, { useState } from 'react';
import type { Meta, StoryObj } from '@storybook/react';

import { Spin } from './Spin';

const meta: Meta<typeof Spin> = {
    title: 'Spin',
    component: Spin,
    argTypes: {
        spinning: { control: 'boolean' },
        children: { control: 'text' },
    },
    args: {
        spinning: true,
    },
};

export default meta;

type Story = StoryObj<typeof Spin>;

export const Basic: Story = {
    render: (args) => <Spin {...args}/>,
};
