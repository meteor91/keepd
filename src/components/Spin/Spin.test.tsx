import React from 'react';
import { render, screen } from '@testing-library/react';
import { Spin } from './Spin';

describe('Spin', () => {
    it('should render correct', () => {
        const { container } = render(
            <Spin spinning={true}>content</Spin>,
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    it('spin spinning', () => {
        const { rerender } = render(
            <Spin spinning={true}>content</Spin>,
        );
        expect(screen.getByTestId('spinner')).toBeInTheDocument();
        rerender(<Spin spinning={false}>content</Spin>);
        expect(screen.queryByTestId('spinner')).not.toBeInTheDocument();
        expect(screen.getByTestId('spin-content')).toHaveTextContent('content');
    });
});
