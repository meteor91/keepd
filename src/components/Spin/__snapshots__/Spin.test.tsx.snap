// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spin should render correct 1`] = `
<div
  class="spinNested"
>
  <div
    class="spin"
    data-testid="spinner"
  >
    <div
      class="loader"
    />
  </div>
  <div
    class="contentBlur"
    data-testid="spin-content"
  >
    content
  </div>
</div>
`;
